nuxt build fails with spa mode

### Version

[v2.10.0](https://github.com/nuxt.js/releases/tag/v2.10.0)

### Reproduction link

[https://github.com/vinayakkulkarni/nuxt-2.10-bug](https://github.com/vinayakkulkarni/nuxt-2.10-bug)

### Steps to reproduce

npm install
npm run build

### What is expected ?

Nuxt to build all routes and files post build success. 

### What is actually happening?

```
ℹ Generating pages                                                                                                                                                                                  18:52:53

 ERROR   /                                                                                                                                                                                          18:52:53

TypeError: Cannot read property 'renderRoute' of undefined
    at Generator.generateRoute (/Users/vinayak/Development/Personal/nuxt/nuxt-2.10-bug/node_modules/@nuxt/generator/dist/generator.js:222:42)
    at /Users/vinayak/Development/Personal/nuxt/nuxt-2.10-bug/node_modules/@nuxt/generator/dist/generator.js:120:24
    at async Promise.all (index 0)
    at async Generator.generateRoutes (/Users/vinayak/Development/Personal/nuxt/nuxt-2.10-bug/node_modules/@nuxt/generator/dist/generator.js:115:7)
    at async Generator.generate (/Users/vinayak/Development/Personal/nuxt/nuxt-2.10-bug/node_modules/@nuxt/generator/dist/generator.js:48:20)
    at async Object.run (/Users/vinayak/Development/Personal/nuxt/nuxt-2.10-bug/node_modules/@nuxt/cli/dist/cli-build.js:101:5)
    at async NuxtCommand.run (/Users/vinayak/Development/Personal/nuxt/nuxt-2.10-bug/node_modules/@nuxt/cli/dist/cli-command.js:2575:7)


 FATAL  Cannot read property 'renderRoute' of undefined                                                                                                                                             18:52:53

  at Generator.afterGenerate (node_modules/@nuxt/generator/dist/generator.js:168:43)
  at async Generator.generate (node_modules/@nuxt/generator/dist/generator.js:50:5)
  at async Object.run (node_modules/@nuxt/cli/dist/cli-build.js:101:5)
  at async NuxtCommand.run (node_modules/@nuxt/cli/dist/cli-command.js:2575:7)


   ╭────────────────────────────────────────────────────────────────╮
   │                                                                │
   │   ✖ Nuxt Fatal Error                                           │
   │                                                                │
   │   TypeError: Cannot read property 'renderRoute' of undefined   │
   │                                                                │
   ╰────────────────────────────────────────────────────────────────╯

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nuxt-2.10-bug@1.0.0 build: `nuxt build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the nuxt-2.10-bug@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/vinayak/.npm/_logs/2019-10-06T13_22_53_417Z-debug.log
```

### Additional comments?

Related issues: #6500, #6513 & #6514

<!--cmty--><!--cmty_prevent_hook-->
<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9867">#c9867</a>)</em></sub></div>

--- a/distributions/nuxt/README.md
+++ b/distributions/nuxt/README.md
@@ -99,7 +99,6 @@ It's as simple as that!
 
 Or you can start by using one of our starter templates:
 
-- [starter](https://github.com/nuxt-community/starter-template): Basic Nuxt.js project template
 - [express](https://github.com/nuxt-community/express-template): Nuxt.js + Express
 - [koa](https://github.com/nuxt-community/koa-template): Nuxt.js + Koa
 - [adonuxt](https://github.com/nuxt-community/adonuxt-template): Nuxt.js + AdonisJS

--- a/packages/cli/src/commands/build.js
+++ b/packages/cli/src/commands/build.js
@@ -62,6 +62,8 @@ export default {
   },
   async run (cmd) {
     const config = await cmd.getNuxtConfig({ dev: false, server: false, _build: true })
+    const shouldGenerate = config.mode === 'spa' && cmd.argv.generate !== false
+    config.server = shouldGenerate
     const nuxt = await cmd.getNuxt(config)
 
     if (cmd.argv.lock) {
@@ -72,15 +74,14 @@ export default {
       }))
     }
 
-    if (nuxt.options.mode !== 'spa' || cmd.argv.generate === false) {
+    if (shouldGenerate) {
+      // Build + Generate for static deployment
+      const generator = await cmd.getGenerator(nuxt)
+      await generator.generate({ build: true })
+    } else {
       // Build only
       const builder = await cmd.getBuilder(nuxt)
       await builder.build()
-      return
     }
-
-    // Build + Generate for static deployment
-    const generator = await cmd.getGenerator(nuxt)
-    await generator.generate({ build: true })
   }
 }

--- a/packages/cli/src/commands/build.js
+++ b/packages/cli/src/commands/build.js
@@ -62,8 +62,7 @@ export default {
   },
   async run (cmd) {
     const config = await cmd.getNuxtConfig({ dev: false, server: false, _build: true })
-    const shouldGenerate = config.mode === 'spa' && cmd.argv.generate !== false
-    config.server = shouldGenerate
+    config.server = config.mode === 'spa' && cmd.argv.generate !== false
     const nuxt = await cmd.getNuxt(config)
 
     if (cmd.argv.lock) {
@@ -74,7 +73,7 @@ export default {
       }))
     }
 
-    if (shouldGenerate) {
+    if (nuxt.options.mode === 'spa' && cmd.argv.generate !== false) {
       // Build + Generate for static deployment
       const generator = await cmd.getGenerator(nuxt)
       await generator.generate({ build: true })

